packages/castmill/lib/castmill/addons/medias/components/index.tsx

```diff
@@ -33,13 +32,36 @@ const MediasPage: Component<{
   store: AddonStore;
   params: any; //typeof useSearchParams;
 }> = (props) => {
-  const toast = useToast();
   const [data, setData] = createSignal<JsonMedia[]>([], {
     equals: false,
   });
 
+  const [teams, setTeams] = createSignal<Team[]>([]);
+  const [selectedTeamId, setSelectedTeamId] = createSignal<number | null>(null);
+
   const itemsPerPage = 10; // Number of items to show per page
 
+  // Fetch teams for the organization
+  createEffect(async () => {
```

Comment:
Same as above, refactor into a service. As the code is mostly the same as above, you should be able to generalise it.